app/models/katello/content_view_component.rb

```diff
@@ -2,24 +2,89 @@ module Katello
   class ContentViewComponent < Katello::Model
     self.include_root_in_json = false
 
-    belongs_to :content_view, :class_name => "Katello::ContentView",
+    belongs_to :composite_content_view, :class_name => "Katello::ContentView",
                               :inverse_of => :content_view_components
     belongs_to :content_view_version, :class_name => "Katello::ContentViewVersion",
                                       :inverse_of => :content_view_components
+    belongs_to :content_view, :class_name => "Katello::ContentView"
 
     validates_lengths_from_database
-    validates :content_view_version_id, :uniqueness => {:scope => :content_view_id}
-    validate :content_view_types
+    validates :content_view_version_id, :uniqueness => {:scope => :composite_content_view_id}, :if => proc { |r| r.content_view_version_id.present? }
+    validates :content_view_id, :uniqueness => {:scope => :composite_content_view_id}, :if => proc { |r| r.content_view_id.present? }
+    validates :composite_content_view, :presence => true
+
+    validate :ensure_valid_attributes
+
+    before_validation :update_content_view
+
+    def latest_version
+      if latest?
+        self.content_view.versions.last
```

Comment:
The `last` method here I think will sort by primary key. This probably won't work if the user has an incremental update (e.g. 2.2) that was published after a major update (e.g. 3). Probably need to sort by version major/minor instead.